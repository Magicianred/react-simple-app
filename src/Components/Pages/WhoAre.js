import React, { Component } from 'react';
import Jumbotron from 'react-bootstrap/Jumbotron';

/**
 * Component for showing the Who Are page.
 * 
 * @component
 * @example
 * return (
 *   <WhoAre />
 * )
 */
class WhoAre extends Component {
    render() {
        return (
            <div className="page">
                <Jumbotron className="page whoare">
                    <h2>Who Are</h2>
                    <p>This is the who are page</p>
                </Jumbotron>
            </div>
        );
    }
}

export default WhoAre;