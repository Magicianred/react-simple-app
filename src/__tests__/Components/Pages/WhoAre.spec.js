import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { create } from 'react-test-renderer';

import WhoAre from "./../../../Components/Pages/WhoAre";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe("WhoAre Component", () => {

    describe("Snapshots", () => {
    
        it("Matches the snapshot (with default title)", () => {
            const instance = create(<WhoAre />);
            expect(instance.toJSON()).toMatchSnapshot();
        });
    
    });

    describe("Check content elements", () => {
        it("Render Component", () => {
            act(() => {
                render(<WhoAre />, container);
            });
            expect(container.querySelector("h2").textContent).toBe("Who Are");
            expect(container.querySelector("p").textContent).toBe("This is the who are page");
        });
    });
});