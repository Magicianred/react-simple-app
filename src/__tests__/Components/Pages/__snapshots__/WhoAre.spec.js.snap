// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhoAre Component Snapshots Matches the snapshot (with default title) 1`] = `
<div
  className="page"
>
  <div
    className="page whoare jumbotron"
  >
    <h2>
      Who Are
    </h2>
    <p>
      This is the who are page
    </p>
  </div>
</div>
`;
